### ISSUE TITLE
Snapping for capturing data

### ISSUE THREAD

**Issue by saberraz** [MerginMaps/input#481]
Allow snapping when capturing features.

**Comment by she-weeds** [comment]
This would be especially good for point features, would love to see this implemented.

**Comment by l2kuami** [comment]
This will be a great improvement which can extend usability of Input in more and more topics.

**Comment by TragicWinter1992** [comment]
Hello, when will this feature be implemented

**Comment by PeterPetrik** [comment]
Tentatively we would like to have it during summer 2022

**Comment by l2kuami** [comment]
Hello, 

I am back to the office now. 

Let me revert to you  ASAP. 

Regards. 


Kuami Layibo Directeur de production
DATAKAD - Groupe GEOLIANCE
ARRIBAT CENTER,Avenue Ibn El Khattab, B4-N4 Cp : 10080, Agdal, Rabat, Maroc
t: +212 537 67 35 68 <tel:+212%20537%2067%2035%2068>m: +212 662 04 49 94 <tel:+212%20662%2004%2049%2094>w: www.kadran-ingenierie.fr / www.datageo.fr <applewebdata://AE3607C4-8C78-4477-B9BA-BCE77A218EE4/www.kadran-ingenierie.fr%20/%20www.datageo.fr>e: ***@***.*** ***@***.***>
 <applewebdata://AE3607C4-8C78-4477-B9BA-BCE77A218EE4/linkedin.com/in/kuami-layibo>

**Comment by TragicWinter1992** [comment]
Hello Luthra,
      That is a very long time to resolution.

Can it be done anytime sooner than later?.

Kind regards
Joshua

**Comment by PeterPetrik** [comment]
@TragicWinter1992 @l2kuami would it be possible to write a few use-cases where snapping is needed in the real on-field situation?

**Comment by TragicWinter1992** [comment]
Sure, as an electric utility, we usually map our electrical  network (assets) as is on the field.


Pole(points) and lines(lines) need to snap to each other.

1. Without it, they'll remain hanging, which is not a very representation.

2. Also, there are some aggregate functions used between assets, with snapping they can't work.

Kind regards 
Joshua

**Comment by l2kuami** [comment]
Hello Sir, 

Got your mails.

Isn’t any default snapping tool as the one in QGIS that can be used in Input? 

Regards. 

Kuami Layibo Directeur de production
DATAKAD - Groupe GEOLIANCE
ARRIBAT CENTER,Avenue Ibn El Khattab, B4-N4 Cp : 10080, Agdal, Rabat, Maroc
t: +212 537 67 35 68 <tel:+212%20537%2067%2035%2068>m: +212 662 04 49 94 <tel:+212%20662%2004%2049%2094>w: www.kadran-ingenierie.fr / www.datageo.fr <applewebdata://440969EA-1A7E-41C1-8AF8-A2E947CC38E7/www.kadran-ingenierie.fr%20/%20www.datageo.fr>e: ***@***.*** ***@***.***>
 <applewebdata://440969EA-1A7E-41C1-8AF8-A2E947CC38E7/linkedin.com/in/kuami-layibo>

**Comment by PeterPetrik** [comment]
The core could be used, but you still need to integrate it into GUI, implement settings in Input settings dialog and make everything work together in various situations (digitizing points, lines, polygons, how to select layer for snapping etc)

**Comment by MTrim** [comment]
Hi!

I agree with @TragicWinter1992:

I have several customers in the utility sector who have to collect lines (water, sewer, gas, telecommunicataion,...) and points (manholes, mains, laterals, valves, distribution boxes,...). The easiest way is to collect points and digitize the lines afterwards. To erase errors it is necessary to digitize these lines in the field. So snapping would be a great enhancement and I am looking forward to this feature.

**Comment by TragicWinter1992** [comment]
Hello Luthra, any progress on this development.

**Comment by PeterPetrik** [comment]
Hi, we haven't started this task yet. Best to check/ping us on slack merginmaps.com/community/join

**Comment by tomasMizera** [comment]
Hello @she-weeds @TragicWinter1992 @MTrim and @l2kuami, snapping is now available in Input version 1.6.0 🙂

**Comment by TragicWinter1992** [comment]
Wow show me

**Comment by l2kuami** [comment]
Wowww.. Great… thank a lot… 

Let test it.. 

Any social confirmation to make before allowing snapping? 

Regards. 

Kuami Layibo Directeur de production
DATAKAD - Groupe GEOLIANCE
ARRIBAT CENTER,Avenue Ibn El Khattab, B4-N4 Cp : 10080, Agdal, Rabat, Maroc
t: +212 537 67 35 68 <tel:+212%20537%2067%2035%2068>m: +212 662 04 49 94 <tel:+212%20662%2004%2049%2094>w: www.kadran-ingenierie.fr / www.datageo.fr <applewebdata://2466660C-0EAB-46A0-BB39-B4DAB75D5DD6/www.kadran-ingenierie.fr%20/%20www.datageo.fr>e: ***@***.*** ***@***.***>
 <applewebdata://2466660C-0EAB-46A0-BB39-B4DAB75D5DD6/linkedin.com/in/kuami-layibo>

**Comment by tomasMizera** [comment]
This is a part of a message we wrote on our community announcement channel, [you can join it to always have the most recent information: join slack.](https://merginmaps.com/community/join) It says how to set it up (we will soon add it to our documentation).

"""
Moreover, we have released **Mergin Maps QGIS plugin 2022.5 where you can allow/disable snapping for MM Input**. In order to change snapping, upgrade to the latest 2022.5 MM Plugin and navigate to Project -> Properties.. -> Mergin Maps -> Snapping. Here you have three options - _no snapping_, _basic snapping_ and _follow QGIS settings_.  
- **no snapping**: MM Input will not use snapping even if the QGIS project uses it (default)
- **basic snapping:** MM Input will use snapping, but with its own configuration, no matter what is set in the QGIS project
- **follow QGIS settings**: MM Input will use the same configuration for snapping as is used in the QGIS project

"""

> **TLDR**: snapping is turned off by default, you need to use the latest version of Mergin Maps QGIS plugin to allow it in QGIS settings.

**Comment by l2kuami** [comment]
I am not sure to understand every thing but, actually there is no way to use snapping in Input without additional development. 

I am right? 
Kuami Layibo Directeur de production
DATAKAD - Groupe GEOLIANCE
ARRIBAT CENTER,Avenue Ibn El Khattab, B4-N4 Cp : 10080, Agdal, Rabat, Maroc
t: +212 537 67 35 68 <tel:+212%20537%2067%2035%2068>m: +212 662 04 49 94 <tel:+212%20662%2004%2049%2094>w: www.kadran-ingenierie.fr / www.datageo.fr <applewebdata://FE963044-FA31-4B84-B57F-EF43A4476FFA/www.kadran-ingenierie.fr%20/%20www.datageo.fr>e: ***@***.*** ***@***.***>
 <applewebdata://FE963044-FA31-4B84-B57F-EF43A4476FFA/linkedin.com/in/kuami-layibo>

**Comment by saberraz** [comment]
You need to set it in the plugin:
https://merginmaps.com/docs/gis/features/#snapping
